File: 1356. Sort Integers by The Number of 1 Bits/solution.java
class Solution {
    public int count(int n){
        int rs  = 0 ; 
        while( n!= 0 ){
            if( ( n & 1 ) == 1 ){
                rs++ ;
            }
            n >>= 1 ; 
        }
        return rs ; 
    }
    public int[] sortByBits(int[] arr) {
        int n = arr.length ; 
        ArrayList<ArrayList<Integer>>  list = new ArrayList<>() ; 
        for(int i=0 ; i<32 ; i++){
            list.add( new ArrayList<>()  ) ;  
        }
        Arrays.sort( arr ) ; 
        for(int i=0 ; i<n ; i++ ){
            list.get( count( arr[i] ) ).add( arr[i] ) ; 
        }
        int current = 0 ;
        for(int i=0 ; i<32 ;i++){
            for(int j=0 ; j< list.get(i).size() ; j++ ){
                arr[current++] = list.get(i).get(j) ; 
            }
        }
        return arr ; 
    }
}

File: 1356. Sort Integers by The Number of 1 Bits/solution.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        sort(arr.begin(), arr.end(), [](int a, int b) {
            int popcountA = __builtin_popcount(a);
            int popcountB = __builtin_popcount(b);
            return popcountA == popcountB ? a < b : popcountA < popcountB;
        });
        return arr;
    }
};
